apm-agent-core/src/main/java/co/elastic/apm/agent/metrics/MetricRegistry.java

```diff
@@ -159,14 +159,30 @@ public DoubleSupplier getGauge(String name, Labels labels) {
         return null;
     }
 
-    public void report(MetricsReporter metricsReporter) {
+    /**
+     * Executes the following steps within a single read-operation critical section:
+     * <ul>
+     *     <li>Switch between active and inactive MetricSet buffers</li>
+     *     <li>Report the inactivated buffer (optional)</li>
+     *     <li>Reset the inactivated buffer</li>
+     * </ul>
+     *
+     * @param metricsReporter a reporter to be used for reporting the inactivated metrics buffer. May be {@code null}
+     *                        if reporting is not required.
+     */
+    public void switchBuffersAndReport(@Nullable MetricsReporter metricsReporter) {
```

Comment:
[minor] maybe call it flip phase and report to be consistent with the reader-writer-phaser terminology? the word `buffer` is not used elsewhere in this class.